File: collie/src/main/java/com/snail/collie/fps/ANRMonitorRunnable.java
package com.snail.collie.fps;

import android.app.Activity;

import java.lang.ref.WeakReference;

public abstract class ANRMonitorRunnable implements Runnable {


    public WeakReference<Activity> activityRef;
    public boolean invalid;

    public ANRMonitorRunnable(WeakReference<Activity> mActivityRef) {
        this.activityRef = mActivityRef;
        this.invalid = false;
    }

}

File: collie/src/main/java/com/snail/kotlin/fps/ANRMonitorRunnable.kt
package com.snail.kotlin.fps

import android.app.Activity
import java.lang.ref.WeakReference

/**
 * Author: snail
 * Data: 2021/10/9.
 * Des:
 * version:
 */
//构造函数是否可以修改成类同名

abstract class ANRMonitorRunnable(var invalid: Boolean, var activityRef: WeakReference<Activity>) : Runnable